projects/packages/videopress/src/client/state/reducers.js

```diff
@@ -269,23 +272,40 @@ const videos = ( state, action ) => {
 			const { id, hasBeenDeleted, video: deletedVideo } = action;
 			const _metaItems = state?._meta?.items || [];
 			const _metaVideo = _metaItems[ id ] || {};
-			const uploadedVideoCount = state.uploadedVideoCount - 1;
+			const removedVideos = [ ...( state.removedVideos ?? [] ) ];
+			const removedVideoIndex = removedVideos.indexOf( id );
 
-			if ( ! _metaVideo ) {
+			if ( ! _metaVideo || removedVideoIndex < 0 ) {
 				return state;
 			}
 
+			removedVideos.splice( removedVideoIndex, 1 );
+
+			const processedAllRemovedVideos = removedVideos.length === 0;
+			let uploadedVideoCount = state.uploadedVideoCount ?? 0;
+			let removedVideoCount = state.removedVideoCount ?? 0;
+
+			if ( ! hasBeenDeleted ) {
+				removedVideoCount -= 1;
+			}
+
+			if ( processedAllRemovedVideos ) {
+				uploadedVideoCount = uploadedVideoCount - removedVideoCount;
+				removedVideoCount = 0;
+			}
+
 			return {
 				...state,
+				removedVideos,
 				uploadedVideoCount,
+				removedVideoCount,
+				processedAllRemovedVideos,
```

Comment:
Looks like this data should be located in the `_meta` field, not? Looks like transient data.